superset/migrations/versions/2023-03-27_12-30_7e67aecbf3f1_chart_ds_constraint.py

```diff
@@ -48,13 +51,16 @@ class Slice(Base):  # type: ignore
 def upgrade_slc(slc: Slice) -> None:
     # clean up all charts with datasource_type not != table
     slc.datasource_type = "table"
+    ds_id = None
+    ds_type = None
     try:
         params_dict = json.loads(slc.params)
         ds_id, ds_type = params_dict["datasource"].split("__")
```

Comment:
```suggestion
        ds_id, ds_type = params_dict["datasource"].split("__")
        # the assumption here is that the query was saved as a dataset
        # but the type wasn't written properly to the slice
        # by updating the type here we expect it will either work
        # or it will 404 when the dataset is looked up.        
```